packages/phishing-controller/src/PhishingController.ts

```diff
@@ -558,13 +606,29 @@ export class PhishingController extends BaseController<
       return;
     }
     const hotlist = hotlistResponse.data;
-    const newPhishingLists = this.state.phishingLists.map((phishingList) =>
-      applyDiffs(
+    const recentlyAdded = requestBlocklistResponse?.recentlyAdded || [];
+    const recentlyRemoved = requestBlocklistResponse?.recentlyRemoved || [];
+
+    const newPhishingLists = this.state.phishingLists.map((phishingList) => {
+      const updatedList = applyDiffs(
         phishingList,
         hotlist,
         phishingListNameKeyMap[phishingList.name],
-      ),
-    );
+      );
+
+      if (
+        phishingList.name === ListNames.MetaMask &&
+        requestBlocklistResponse
+      ) {
+        updatedList.requestBlocklist = updateRequestBlocklist(
```

Comment:
Why are we calculating the diff here instead of in the `applyDiffs` abstraction?

Reply:
The reason is that `applyDiffs` handles diffs differently than how we handle the C2 domain blocklist diffs. We'll need to refactor much of this once openCTI is finished to follow the same pattern as the C2 domain blocklist diffs.

Reply:
I am not sure I follow.. because on the applyDiffs we are already enriching it with the [c2 domain blocklist](https://github.com/MetaMask/core/pull/4526/files#diff-e5501df0e3c3dd854304c4ed16cfecee16fb136cbdc3e8537a224e20c89bed78R86). So why not encapsulate the this [logic here](https://github.com/MetaMask/core/pull/4526/files#diff-73fbecd61d9f85b57fee9c8377652f5563d90e026e54fa105ca91fcc2cbc076fR616-R625) within the applyDiffs method? What am I missing? 🤔